libethash-cl/CLMiner.cpp

```diff
@@ -103,9 +103,11 @@ void CLMiner::report(uint64_t _nonce, WorkPackage const& _w)
 	// TODO: Why re-evaluating?
 	Result r = EthashAux::eval(_w.seed, _w.header, _nonce);
 	if (r.value < _w.boundary)
-		farm.submitProof(Solution{_nonce, r.mixHash, _w.header, _w.seed, _w.boundary});
-	else
-		cwarn << "Invalid solution";
+		farm.submitProof(Solution{_nonce, r.mixHash, _w.header, _w.seed, _w.boundary, false});
```

Comment:
Do we don't know if the result is stale here? (did not look to much at the code itself)

Reply:
No we don't. Submit it as valid. Don't think CL collects the same way... it doesn't seem to use the same pause, reset strategy.

Reply:
The mining thread should not know (and not care) if the solution is stale already. So having a "stale" flag in Solution is not the best design. It is up to stratum thread to figure out what to do with the solution. However, I'm ok with accepting this a temporary solution. I'm working on simplifying CUDAMiner to be similar to CLMiner. Should be single file in the end.

Reply:
There is a fundamental problem with the CUDA GPU code the CL code doesn't have. It can generate stale solutions, whereas I don't think the CL GPU code can. Not sure what your plan is for handling it while maintaining the strict functional boundary you propose.